From 4fc11130cf2c8b89c55d7c77629ca639438e34c8 Mon Sep 17 00:00:00 2001
From: satya kottisa <fvgm76@motorola.com>
Date: Wed, 7 Oct 2015 15:11:10 -0500
Subject: [PATCH 410/959] IKMMINTG-6037: arm/dt: 8952: Add ov16860 Rear camera

Add device tree node needed for 16M ov16860
Rear camera powerup/probe

Change-Id: I8b3915cfb9da32a3ba3263cdfdbe22316966b585
Signed-off-by: satya kottisa <fvgm76@motorola.com>
Reviewed-on: http://gerrit.mot.com/794131
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Sumit Jamadar <xkbr38@motorola.com>
Reviewed-by: Elena Satraitis <w04888@motorola.com>
Reviewed-by: Christopher Fries <cfries@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../boot/dts/qcom/msm8952-camera-sensor-moto.dtsi  | 33 ++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
index bb03507ae7f..14b38f9b980 100644
--- a/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-camera-sensor-moto.dtsi
@@ -23,6 +23,39 @@
 };
 
 &cci {
+	qcom,camera@0 {
+		cell-index = <0>;
+		compatible = "qcom,camera";
+		qcom,csiphy-sd-index = <0>;
+		qcom,csid-sd-index = <0>;
+		qcom,mount-angle = <90>;
+		qcom,led-flash-src = <&led_flash0>;
+		cam_vdig-supply = <&pm8950_l23>;
+		cam_vio-supply = <&pm8950_l10>;
+		qcom,cam-vreg-name = "cam_vdig", "cam_vio";
+		qcom,cam-vreg-min-voltage = <1200000 1800000> ;
+		qcom,cam-vreg-max-voltage = <1200000 1800000>;
+		qcom,cam-vreg-op-mode = <1200000 1800000>;
+		pinctrl-names = "cam_default", "cam_suspend";
+		pinctrl-0 = <&cam_sensor_mclk0_default
+			     &cam_sensor_rear_default>;
+		pinctrl-1 = <&cam_sensor_mclk0_sleep &cam_sensor_rear_sleep>;
+		gpios = <&msm_gpio 26 0>,
+			<&msm_gpio 36 0>;
+		qcom,gpio-reset = <1>;
+		qcom,gpio-req-tbl-num = <0 1>;
+		qcom,gpio-req-tbl-flags = <1 0>;
+		qcom,gpio-req-tbl-label = "CAMIF_MCLK0",
+					  "CAM_RESET0";
+		qcom,sensor-position = <0>;
+		qcom,sensor-mode = <0>;
+		qcom,cci-master = <0>;
+		status = "ok";
+		clocks = <&clock_gcc clk_mclk0_clk_src>,
+			 <&clock_gcc clk_gcc_camss_mclk0_clk>;
+		clock-names = "cam_src_clk", "cam_clk";
+};
+
 	qcom,camera@1 {
 		cell-index = <1>;
 		compatible = "qcom,camera";
-- 
2.11.0

